crates/gitbutler-tauri/src/watcher/handler/calculate_deltas.rs

```diff
@@ -0,0 +1,178 @@
+                    session_id,
+                    path,
+                    std::slice::from_ref(&new_delta),
+                )
+                .context("failed to index deltas")?;
+                self.emit_app_event(&crate::events::Event::deltas(
+                    project_id,
+                    session_id,
+                    std::slice::from_ref(&new_delta),
+                    path,
+                ))?;
+                Ok(true)
+            };
+            Ok((process, session))
+        };
+        let num_paths = paths.len();
+        let num_no_delta = std::thread::scope(|scope| -> Result<usize> {
+            let num_threads = std::thread::available_parallelism()
+                .unwrap_or(NonZeroUsize::new(1).unwrap())
+                .get()
+                .min(paths.len());
+            let mut num_no_delta = 0;
+            let current_session = if num_threads < 2 {
+                let (process, session) = make_processor()?;
+                for path in paths {
+                    if !process(path.as_path())? {
+                        num_no_delta += 1;
+                    }
+                }
+                session
+            } else {
+                let (threads, tx) = {
+                    let (tx, rx) = crossbeam_channel::bounded::<PathBuf>(num_threads);
+                    let threads: Vec<_> = (0..num_threads)
+                        .map(|id| {
+                            std::thread::Builder::new()
+                                .name(format!("gitbutler_delta_thread_{id}"))
+                                .stack_size(512 * 1024)
+                                .spawn_scoped(scope, {
+                                    let rx = rx.clone();
+                                    || -> Result<usize> {
+                                        let mut num_no_delta = 0;
+                                        let (process, _) = make_processor()?;
+                                        for path in rx {
+                                            if !process(path.as_path())? {
+                                                num_no_delta += 1;
+                                            }
+                                        }
+                                        Ok(num_no_delta)
+                                    }
+                                })
+                                .expect("worker thread can be created")
+                        })
+                        .collect();
+                    (threads, tx)
+                };
+                for path in paths {
+                    tx.send(path).expect("many receivers");
+                }
+                drop(tx);
+
+                for thread in threads {
+                    num_no_delta += thread.join().unwrap()?;
+                }
+                let (_, session) = make_processor()?;
+                session
+            };
+            self.index_session(project_id, &current_session)?;
```

Comment:
Note that this call was previously between line 98 and 99, and thus called per `path`. Thus it seems like a waste to call it so often as it doesn't depend on it. This app seems to work fine like this, but there may be side-effects I am unaware of, particularly because the delta-computation doesn't seem to be displayed in the UI.

Reply:
Can always adjust if something isn't right.